debezium-core/src/main/java/io/debezium/transforms/ByLogicalTableRouter.java

```diff
@@ -107,78 +86,154 @@
             // Default is prefixed with "__dbz__" to minimize the likelihood of a conflict with an existing key field name.
             .withDefault("__dbz__physicalTableIdentifier")
             .withDescription("Each record's key schema will be augmented with this field name. The purpose of this " +
-                    "field is to distinguish the different physical tables that can now share a single topic. Make " +
-                    "sure not to configure a field name that is at risk of conflict with existing key schema field " +
-                    "names.");
+                "field is to distinguish the different physical tables that can now share a single topic. Make " +
+                "sure not to configure a field name that is at risk of conflict with existing key schema field " +
+                "names.");
+    private static final Field KEY_FIELD_REPLACEMENT = Field.create("key.field.replacement")
+            .withDisplayName("Key field replacement")
+            .withType(ConfigDef.Type.STRING)
+            .withWidth(ConfigDef.Width.LONG)
+            .withImportance(ConfigDef.Importance.LOW)
+            .withValidation(ByLogicalTableRouter::validateKeyFieldReplacement)
+            .withDescription("The replacement string used in conjunction with " + KEY_FIELD_REGEX.name() +
+                ". This will be used to create the physical table identifier in the record's key.");
+
+    private static final Logger logger = LoggerFactory.getLogger(ByLogicalTableRouter.class);
```

Comment:
It's a pity that all the "clean-up" is not in a separate commit. The changes themselves are good, but having all in one commit make it much harder to review the actual change.